distributed/scheduler.py

```diff
@@ -4087,6 +4087,9 @@ def update_graph(
                     continue
                 ts._host_restrictions = set()
                 ts._worker_restrictions = set()
+                # Make sure `v` is a collection and not a single worker name / address
+                if not isinstance(v, (list, tuple, set)):
+                    v = [v]
```

Comment:
nitpick: We should have enough control about the other end to ensure this is always a tuple, don't we? I'm wondering if this is smth we rather want to do on the other end, i.e. keep code and types clean in this function since it is already quite large and complicated

Reply:
We can do it when [we pack the annotations](https://github.com/dask/dask/blob/0130bd15cde5f0e244e3e9a0dbfab1a8471699ad/dask/highlevelgraph.py#L167) but I am not sure that is a good idea.  The annotation packing is very generic and doesn't know anything about the _meaning_ of the annotations.

Reply:
Note that some checking is already done client-side: https://github.com/dask/dask/blob/main/dask/base.py#L91-L102

Reply:
True, but it is only when we unpack/expand the annotation that we know if the `callable` is indeed returning a list of workers.